src/runtime-rs/crates/agent/src/sock/unix_sock.rs

```diff
@@ -8,37 +8,41 @@ use std::path::Path;
 
 use anyhow::{anyhow, Context, Result};
 use async_trait::async_trait;
+use slog::{info, trace};
 use tokio::{io::Interest, net::UnixStream};
 
 use super::{ConnectConfig, Sock, Stream};
 
+/// Unix socket connection logic for both remote and unix schemes
 #[derive(Debug, PartialEq)]
-pub struct Remote {
+pub struct UnixSock {
     path: String,
+    scheme: &'static str,
 }
 
-impl Remote {
-    pub fn new(path: String) -> Self {
-        Self { path }
+impl UnixSock {
+    pub fn new(path: String, scheme: &'static str) -> Self {
```

Comment:
This is just an observation, so feel free to disregard, but I think I'd expect the scheme as the first argument and the path as the second as `unix://console.sock` -> `UnixSock::new("unix", "console.sock")` feels more natural than `UnixSock::new("console.sock", "unix")`?